fs/artifact_fetcher.go

```diff
@@ -68,6 +71,70 @@ func newArtifactFetcher(refspec reference.Spec, localStore store.BasicStore, rem
 	}, nil
 }
 
+// This is a wrapper for the ORAS remote repository.
+// We only need this to overwrite the Resolve call.
+// By default ORAS will attempt to resolve manifests,
+// so this allows us to resolve layers instead.
+// However, ORAS uses a HEAD request to get layer info,
+// which is not allowed in some repos, so we also
+// add a manual retry with a GET call should we
+// get a 401 or 403 error.
+type orasBlobStore struct {
+	*remote.Repository
+}
+
+func newRemoteBlobStore(r *remote.Repository) *orasBlobStore {
+	return &orasBlobStore{r}
+}
+
+// Logic mostly taken from oras-go. Try to resolve with a HEAD, then a GET request.
+// https://github.com/oras-project/oras-go/blob/d51a392ff5432a9090c64ffec6ca6a8690b55e18/registry/remote/repository.go#L944
+func (r *orasBlobStore) Resolve(ctx context.Context, reference string) (ocispec.Descriptor, error) {
+	ref, err := registry.ParseReference(reference)
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+	refDigest, err := ref.Digest()
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+
+	tr := &clientWrapper{r.Client}
+	buildScheme := "https"
```

Comment:
nit: could be const

Reply:
we shouldn't hardcode `https` here. Use [this logic](https://github.com/awslabs/soci-snapshotter/blob/main/service/resolver/registry.go#L202-L207)